packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/singleton.tsp

```diff
@@ -0,0 +1,173 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Azure.ResourceManager.Models.Resources;
+
+@singleton("default")
+model SingletonTrackedResource is TrackedResource<SingletonTrackedResourceProperties> {
+  @key("singletonTrackedResourceName")
+  @path
+  @segment("singletonTrackedResources")
+  @doc("arm resource name for path")
+  @pattern("^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$")
+  name: string;
+}
```

Comment:
please use the latest template syntax: https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/rules/resource-name-pattern

Reply:
Fixed in the new version.

Reply:
@v-hongli1 Fix seems wrong. Replace `name` with `ResourceNameParameter` in `SingletonTrackedResource`, instead of other properties in `SingletonTrackedResourceProperties`.

Reply:
The new version commit: https://github.com/Azure/cadl-ranch/pull/702/commits/6892aa342c8b347a60897f64cc13990bc2582208